ultravox/training/ddp_utils.py

```diff
@@ -43,3 +47,18 @@ def sharded_iterator(ds: data.IterableDataset, num_shards: int, shard_index: int
     for i, sample in enumerate(ds):
         if i % num_shards == shard_index:
             yield sample
+
+
+def sharded_batch_iterator(
+    ds: data.IterableDataset, batch_size: int, num_shards: int, shard_index: int
+):
+    batch = []
+    for idx, sample in enumerate(ds):
+        if idx % num_shards == shard_index:
+            batch.append((idx, sample))
+            if len(batch) == batch_size:
+                yield batch
+                batch = []
+    # Yield any remaining samples in the last incomplete batch
+    if batch:
+        yield batch
```

Comment:
Why copy & paste this? The same functionality can be used when `batch_size` equals 1. Also I believe the only reference to `sharded_iterator` is already removed.